src/esp/gfx/BatchRenderer.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_GFX_BATCH_RENDERER_H_
+#define ESP_GFX_BATCH_RENDERER_H_
+
+#include <Corrade/Containers/Pointer.h>
+#include <Magnum/GL/GL.h>
+#include <Magnum/Magnum.h>
+#include <cstddef>
+
+namespace esp {
+namespace gfx {
+
+enum class BatchRendererFlag {
+  NoTextures = 1 << 0
+  // TODO memory-map
+};
+typedef Corrade::Containers::EnumSet<BatchRendererFlag> BatchRendererFlags;
+CORRADE_ENUMSET_OPERATORS(BatchRendererFlags)
+
+struct BatchRendererConfiguration {
+  explicit BatchRendererConfiguration();
+  ~BatchRendererConfiguration();
+
+  BatchRendererConfiguration& setFlags(BatchRendererFlags flags);
+  BatchRendererConfiguration& setTileSizeCount(
+      const Magnum::Vector2i& tileSize,
+      const Magnum::Vector2i& tileCount);
+
+  struct State;
+  Corrade::Containers::Pointer<State> state;
```

Comment:
Yes, reducing build times. Here in case of the configuration class it's possibly taken a bit too far and wouldn't need to be there as the state is tiny, but I have this so ingrained in my muscle memory that I used it everywhere. 

For the `BatchRenderer` classes I definitely want to keep it this way because otherwise the header gets way too heavy for the minimal interface it exposes, especially when the alternative Vulkan backend gets added. For the `BatchRendererConfiguration` I can remove that if you feel it's silly -- let me know.

(Related: it's been a while since I worked with the Habitat codebase last time, and I was surprised by how long everything takes to build. I may open a few PRs soon to clean some of that up, because it shouldn't be that an incremental build of the `gfx` library takes more time than rebuilding all of Magnum. I'm not used to so much waiting :sweat_smile:)